qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -721,3 +798,56 @@ def autorange_once() -> bool:
         while autorange_once() and sets < max_changes:
             sets += 1
             time.sleep(self.time_constant())
+
+    def set_sweep_parameters(self,
+                             sweep_param: Parameter,
+                             start: float,
+                             stop: float,
+                             n_points: int = 10,
+                             label: Union[str, None] = None) -> None:
+
+        self.sweep_setpoints.sweep_array = np.linspace(start, stop, n_points)
+        self.sweep_setpoints.unit = sweep_param.unit
+        if label is not None:
+            self.sweep_setpoints.label = label
+        elif sweep_param.label is not None:
+            self.sweep_setpoints.label = sweep_param.label
+
+
+class GeneratedSetPoints(Parameter):
+    """
+    A parameter that generates a setpoint array from start, stop and num points
+    parameters.
+    """
+    def __init__(self,
+                 sweep_array: Iterable[Union[float, int]] = np.linspace(0, 1, 10),
+                 *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.sweep_array = sweep_array
+        self.update_units_if_constant_sample_rate()
+
+    def update_units_if_constant_sample_rate(self) -> None:
+        """
+        If the buffer is filled at a constant sample rate,
+        update the unit to "s" and label to "Time";
+        otherwise do nothing
+        """
+        assert isinstance(self.root_instrument, SR830)
+        SR = self.root_instrument.buffer_SR.get()
+        if SR != 'Trigger':
+            self.unit = 's'
+            self.label = 'Time'
+
+    def set_raw(self, value: Iterable[Union[float, int]]) -> None:
```

Comment:
There is no need to use `Union[float,int]` that is fully equivalent to `float` as int is considered a subtype of float by mypy